core/controllers/reader.py

```diff
@@ -1011,3 +1011,27 @@ def post(self, exploration_id):
             self.payload.get('report_text'),
             self.user_id)
         self.render_json(self.values)
+
+
+class QuestionPlayerHandler(base.BaseHandler):
+    """Provides questions with given skill ids."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
```

Comment:
I'm not sure what kind of decorator to use here
@vinitamurthi

Reply:
I think you can use the can_access_topic_viewer_page acl decorator - https://github.com/oppia/oppia/blob/e647b691c04831e228f1da2bdfd9fce72b76dcb3/core/controllers/acl_decorators.py#L2314